### PR TITLE
Dto

### PR DESCRIPTION
**Pull request by layzimayner**

### CODE REVIEW

**Comment by layzimayner** [comment]
i don't have old version to merge

File: pom.xml [hunk 1]
```diff
@@ -45,6 +47,27 @@
...
+		</dependency>
+		<dependency>
+			<groupId>org.mapstruct</groupId>
+			<artifactId>mapstruct</artifactId>
+			<version>${mapstruct.version}</version>
+		</dependency>
+
+		<dependency>
+			<groupId>org.springframework.data</groupId>
+			<artifactId>spring-data-jpa</artifactId>
```
Comment (Elena-Bruyako) [inline pom.xml]:
```suggestion
			<artifactId>spring-boot-starter-data-jpa</artifactId>
```

File: pom.xml [hunk 2]
```diff
@@ -45,6 +47,27 @@
...
+			<version>3.3.5</version>
```
Comment (Elena-Bruyako) [inline pom.xml]:
```suggestion
```

File: pom.xml [hunk 3]
```diff
@@ -45,6 +47,27 @@
 			<artifactId>lombok</artifactId>
 			<optional>true</optional>
 		</dependency>
+		<dependency>
+			<groupId>org.springframework</groupId>
+			<artifactId>spring-web</artifactId>
+			<version>6.1.14</version>
```
Comment (Elena-Bruyako) [inline pom.xml]:
```suggestion
			<artifactId>spring-boot-starter-web</artifactId>
```

File: pom.xml [hunk 4]
```diff
@@ -54,6 +77,31 @@
 
 	<build>
 		<plugins>
+			<plugin>
+				<groupId>org.apache.maven.plugins</groupId>
+				<artifactId>maven-compiler-plugin</artifactId>
+				<configuration>
+					<source>17</source>
+					<target>17</target>
```
Comment (Elena-Bruyako) [inline pom.xml]:
```suggestion
					<source>${java.version}</source>
					<target>${java.version}</target>
```

File: pom.xml [hunk 5]
```diff
@@ -54,6 +77,31 @@
...
+					<annotationProcessorPaths>
+						<path>
+							<groupId>org.projectlombok</groupId>
+							<artifactId>lombok</artifactId>
+							<version>1.18.34</version>
```
Comment (Elena-Bruyako) [inline pom.xml]:
```suggestion
							<version>${lombok.version}</version>
```

File: src/main/java/com/example/demo/controller/BookController.java [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+import java.util.List;
+
+@RequiredArgsConstructor
+@RestController
+@RequestMapping(value = "/books")
```
Comment (Elena-Bruyako) [inline src/main/java/com/example/demo/controller/BookController.java]:
```suggestion
@RequestMapping("/books")
```

File: src/main/java/com/example/demo/dto/CreateBookRequestDto.java [hunk 7]
```diff
@@ -0,0 +1,13 @@
+package com.example.demo.dto;
+
+import java.math.BigDecimal;
+
+public class CreateBookRequestDto {
+    private Long id;
```
Comment (Elena-Bruyako) [inline src/main/java/com/example/demo/dto/CreateBookRequestDto.java]:
```suggestion
```

File: src/main/java/com/example/demo/dto/BookDto.java [hunk 8]
```diff
@@ -0,0 +1,13 @@
+package com.example.demo.dto;
+
+import java.math.BigDecimal;
+
+public class BookDto {
```
Comment (Elena-Bruyako) [inline src/main/java/com/example/demo/dto/BookDto.java:7]:
```suggestion
@Data
public class BookDto {
```

File: src/main/java/com/example/demo/dto/CreateBookRequestDto.java [hunk 9]
```diff
@@ -0,0 +1,13 @@
+package com.example.demo.dto;
+
+import java.math.BigDecimal;
+
+public class CreateBookRequestDto {
```
Comment (Elena-Bruyako) [inline src/main/java/com/example/demo/dto/CreateBookRequestDto.java:7]:
```suggestion
@Data
public class CreateBookRequestDto {
```

File: src/main/java/com/example/demo/repository/BookRepositoryImpl.java [hunk 10]
```diff
@@ -0,0 +1,59 @@
...
+            throw new DataProcessingException("Can't get books from DB", e);
+        }
+    }
+
+    @Override
+    public Optional<Book> findById(Long id) {
+        try (Session session = sessionFactory.openSession()) {
+            Book book = session.find(Book.class, id);
+            return Optional.ofNullable(book);
+        }
```
Comment (Elena-Bruyako) [inline src/main/java/com/example/demo/repository/BookRepositoryImpl.java:59]:
```suggestion
        } catch (Exception e) {
            throw new DataProcessingException(.....
        }
```

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
see comments

**Review by Elena-Bruyako — APPROVED** [review]

**Pull request merged**